pandas/tests/frame/test_join.py

```diff
@@ -0,0 +1,158 @@
+
+
+class TestDataFrameJoin(TestData):
+
+    def test_join(self):
+        df1 = pd.DataFrame({'a': [20, 10, 0]}, index=[2, 1, 0])
+        df2 = pd.DataFrame({'b': [100, 200, 300]}, index=[1, 2, 3])
+
+        result = df1.join(df2)
+        expected = pd.DataFrame({'a': [20, 10, 0], 'b': [200, 100, None]},
+                                index=[2, 1, 0])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='left')
+        expected = pd.DataFrame({'a': [20, 10, 0], 'b': [200, 100, None]},
+                                index=[2, 1, 0])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='right')
+        expected = pd.DataFrame({'a': [10, 20, None], 'b': [100, 200, 300]},
+                                index=[1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='inner')
+        expected = pd.DataFrame({'a': [20, 10], 'b': [200, 100]},
+                                index=[2, 1])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='outer')
+        expected = pd.DataFrame({'a': [0, 10, 20, None],
+                                 'b': [None, 100, 200, 300]},
+                                index=[0, 1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+    def test_join_sort(self):
+        df1 = pd.DataFrame({'a': [20, 10, 0]}, index=[2, 1, 0])
+        df2 = pd.DataFrame({'b': [100, 200, 300]}, index=[1, 2, 3])
+
+        result = df1.join(df2, sort=True)
+        expected = pd.DataFrame({'a': [0, 10, 20], 'b': [None, 100, 200]},
+                                index=[0, 1, 2])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='left', sort=True)
+        expected = pd.DataFrame({'a': [0, 10, 20], 'b': [None, 100, 200]},
+                                index=[0, 1, 2])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='right', sort=True)
+        expected = pd.DataFrame({'a': [10, 20, None], 'b': [100, 200, 300]},
+                                index=[1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+        result = df2.join(df1, how='right', sort=True)
+        expected = pd.DataFrame([[None, 0], [100, 10], [200, 20]],
+                                columns=['b', 'a'], index=[0, 1, 2])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='inner', sort=True)
+        expected = pd.DataFrame({'a': [10, 20], 'b': [100, 200]},
+                                index=[1, 2])
+        tm.assert_frame_equal(result, expected)
+
+        result = df1.join(df2, how='outer', sort=True)
+        expected = pd.DataFrame({'a': [0, 10, 20, None],
+                                 'b': [None, 100, 200, 300]},
+                                index=[0, 1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+    def test_join_index(self):
```

Comment:
all of these tests on index should be in ``tests/pandas/tests/indexes/test_base.py``. These are done pretty generically, follow the existing pattern (also wouldn't hurt to move these tests and your new ones to a new test class).